internal/venice-common/src/main/java/com/linkedin/venice/writer/update/UpdateBuilderImpl.java

```diff
@@ -121,26 +127,49 @@ public GenericRecord build() {
     return updateRecord;
   }
 
-  private Optional<Exception> validateUpdateRecordIsSerializable(GenericRecord updateRecord) {
+  private Exception validateUpdateRecordIsSerializable(GenericRecord updateRecord) {
     try {
       serializer.serialize(updateRecord);
     } catch (Exception serializationException) {
-      return Optional.of(serializationException);
+      return serializationException;
     }
-    return Optional.empty();
+    return null;
   }
 
   /**
-   * Given a field from the partial update schema and find the type of its corresponding value field.
+   * Given a field from the partial update schema and find the schema of its corresponding value field.
    *
    * @param updateField A field from the partial update schema.
-   * @return Type of its corresponding value field.
+   * @return Schema of its corresponding value field.
    */
-  private Schema.Type getCorrespondingValueFieldType(Schema.Field updateField) {
+  private Schema getCorrespondingValueFieldSchema(String updateFieldName) {
+    // Each field in partial update schema is a union of multiple schemas.
+    List<Schema> updateFieldSchemas = updateRecord.getSchema().getField(updateFieldName).schema().getTypes();
+    // The last schema in the union is the schema of the field in the corresponding value schema.
+    return updateFieldSchemas.get(updateFieldSchemas.size() - 1);
+  }
+
+  /**
+   * Given a field from the partial update schema and find the schema of its corresponding value field.
+   *
+   * @param updateField A field from the partial update schema.
+   * @return Schema of its corresponding value field.
+   */
+  private Schema getCorrespondingValueFieldSchema(Schema.Field updateField) {
     // Each field in partial update schema is a union of multiple schemas.
     List<Schema> updateFieldSchemas = updateField.schema().getTypes();
     // The last schema in the union is the schema of the field in the corresponding value schema.
-    return updateFieldSchemas.get(updateFieldSchemas.size() - 1).getType();
+    return updateFieldSchemas.get(updateFieldSchemas.size() - 1);
```

Comment:
Can we get rid of duplicate logic between these two?
For example:
```
private Schema getCorrespondingValueFieldSchema(String updateFieldName) {
    return getCorrespondingValueFieldSchema(updateRecord.getSchema().getField(updateFieldName));
  }
```

Reply:
Sure can. I think that was the original intention, but I missed the consolidating into one function. Will update it. Thanks